src/PointwiseFunctions/AnalyticSolutions/Xcts/ConstantDensityStar.cpp

```diff
@@ -0,0 +1,149 @@
+#include "Utilities/MakeWithValue.hpp"
+// IWYU pragma: no_forward_declare Tensor
+
+namespace {
+
+// Find the alpha parameter that corresponds to the weak-field solution,
+// since this is the solution we get when we set \psi = 1 initially
+double compute_alpha(const double density, const double radius) noexcept {
+  const double alpha_source = sqrt(2 * M_PI * density / 3.) * radius;
+  return RootFinder::newton_raphson(
+      [alpha_source](const double a) noexcept {
+        const double a_square = pow<2>(a);
+        const double pow_2_one_plus_a_square = pow<2>(1. + a_square);
+        return std::pair<double, double>{
+            alpha_source - a_square * pow<3>(a) /
+                               (pow_2_one_plus_a_square * (1. + a_square)),
+            pow<2>(a_square) * (a_square - 5.) /
+                pow<2>(pow_2_one_plus_a_square)};
+      },
+      // TODO: Choose better initial guess?
+      2. * sqrt(5.), sqrt(5.), std::numeric_limits<double>::max(),
+      // TODO: choose useful precision (14 digits?)
+      14);
+}
+
+template <typename DataType>
+Scalar<DataType> compute_piecewise(
+    const tnsr::I<DataType, 3, Frame::Inertial>& x, const double radius,
+    const double inner_value, const double outer_value) noexcept {
+  const DataVector r = get(magnitude(x));
+  return Scalar<DataType>(inner_value - (inner_value - outer_value) *
+                                            step_function(r - radius));
+}
+
+}  // namespace
+
+namespace Xcts {
+namespace Solutions {
+
+ConstantDensityStar::ConstantDensityStar(const double density,
+                                         const double radius,
+                                         const OptionContext& context)
+    : density_(density), radius_(radius) {
+  const double critical_density =
+      3. * pow<5>(5) / (2. * pow<6>(6) * M_PI * pow<2>(radius));
+  if (density <= critical_density) {
+    alpha_ = compute_alpha(density, radius);
+  } else {
+    PARSE_ERROR(context,
+                "A ConstantDensityStar has no solutions for a density below "
+                "the critical density ("
+                    << critical_density << ").");
+  }
+}
+
+void ConstantDensityStar::pup(PUP::er& p) noexcept {
+  p | density_;
+  p | radius_;
+  alpha_ = compute_alpha(density_, radius_);
+}
+
+tuples::TaggedTuple<Xcts::Tags::ConformalFactor<DataVector>>
+ConstantDensityStar::variables(
+    const tnsr::I<DataVector, 3, Frame::Inertial>& x,
+    tmpl::list<Xcts::Tags::ConformalFactor<DataVector>> /*meta*/) const
+    noexcept {
+  const DataVector r = get(magnitude(x));
+  const double inner_prefactor =
+      sqrt(alpha_ * radius_) / std::pow(2. * M_PI * density_ / 3., 0.25);
+  const double alpha_times_radius_square = square(alpha_ * radius_);
+  const double beta = inner_prefactor / sqrt(1. + square(alpha_)) - radius_;
+  Scalar<DataVector> conformal_factor{r.size()};
+  for (size_t i = 0; i < r.size(); i++) {
+    if (r[i] <= radius_) {
```

Comment:
Is this a place where LIKELY/UNLIKELY can be used?

Reply:
I don’t know.. @nilsdeppe can you answer?

Reply:
I can ask the question that will help answer :)
- In general, do you expect the condition to be true or false more often? If you are unsure or expect them to be approximately the same then leave it as is. If you know that say 90% of the time it will be true, then use `if (LIKELY(r[i] <= radius_))`. If you know that 90% of the time it will be true, use `UNLIKELY` :)

Reply:
Then the macros do not apply here. Thanks